src/main/scala/thurloe/database/ThurloeDatabaseConnector.scala

```diff
@@ -212,14 +239,17 @@ case object ThurloeDatabaseConnector extends DataAccess with LazyLogging {
    * @return The type of operation which was carried out (as a Future)
    */
   private def databaseWrite(userKeyValuePair: UserKeyValuePair,
-                            encryptedValue: EncryptedBytes,
-                            samDao: SamDAO): Future[DatabaseOperation] = {
-    val lookupExists = lookupIncludingDatabaseId(userKeyValuePair.userId, userKeyValuePair.keyValuePair.key, samDao)
```

Comment:
databaseWrite first does a lookup in the database to decide whether to do an update or insert (not sure why since postgres does upserts). `lookupIncludingDatabaseId` includes the call to sam and then looks up the key in thurloe for any of the types of user ids returned by sam for the user. I decided to switch to just doing the lookup for the specific userid-key combo because the behavior is simpler and easier to reason about. 

In the previous setup if someone had b2cId1-key1 written and then tried to write subjectId-key1 thurloe would behave as if subjectId-key1 was already written (if sam said it belonged to the same user) and return OK response instead of Created. I am not sure this has caused any problems but I think it could have.